src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs

```diff
@@ -256,6 +256,16 @@ void Model_PropertyChanged(object sender, System.ComponentModel.PropertyChangedE
                     RaisePropertyChanged("RunEnabled");
                     RaisePropertyChanged("RunButtonEnabled");
                     RaisePropertyChanged("RunButtonToolTip");
+                    if (Application.Current != null)
+                    {
+                        Application.Current.Dispatcher.Invoke(new Action(() =>
+                        {
+                            dynamoViewModel.ShowOpenDialogAndOpenResultCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.NewHomeWorkspaceCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.OpenRecentCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.CloseHomeWorkspaceCommand.RaiseCanExecuteChanged();
+                        }));
+                    }
```

Comment:
Yep okay that makes sense. We now know `RaisePropertyChanged` automatically handle cross-thread notifications, I was expecting `RaiseCanExecuteChanged` to do the same, but it seems like that isn't the case? Did you explicitly see issue with cross-thread exception if you call `RaiseCanExecuteChanged` directly on different thread? Just being curious...